consensus/phases/thirdphase.go

```diff
@@ -129,6 +131,36 @@ func (tp *ThirdPhaseImpl) Execute(ctx context.Context, pulse *core.Pulse, state
 	}, nil
 }
 
+func (tp *ThirdPhaseImpl) handleJoinClaims(ref core.RecordRef, state *SecondPhaseState, entropy core.Entropy) {
+	handler := claimhandler.NewJoinHandler(len(tp.NodeKeeper.GetActiveNodes()))
+	claims := state.UnsyncList.GetClaims(ref)
+
+	if len(claims) == 0 {
+		return
+	}
+
+	joinClaims := make([]*packets.NodeJoinClaim, 0)
+	resClaims := make([]packets.ReferendumClaim, 0)
+	logger := inslogger.FromContext(context.Background())
+	logger.Debugf("[ handleJoinClaims ] old claims count: %d", len(claims))
+	for _, claim := range claims {
+		c, ok := claim.(*packets.NodeJoinClaim)
+		if !ok {
+			resClaims = append(resClaims, c)
+			continue
+		}
+		joinClaims = append(joinClaims, c)
+	}
+	updatedJoinClaims := handler.HandleClaims(joinClaims, entropy)
```

Comment:
We select join claims by each node instead of all join claims by all nodes. It is a big defect.
For example, we have 5 nodes, each has 1 claim. Each time here you will get 1 claim, because 1 < 5/3. So total claim count would be 5 instead of 1

Reply:
done